src/validation.cpp

```diff
@@ -3120,8 +3120,8 @@ static uint64_t CalculateCurrentUsage()
 /* Prune a block file (modify associated database entries)*/
 void PruneOneBlockFile(const int fileNumber)
 {
-    for (BlockMap::iterator it = mapBlockIndex.begin(); it != mapBlockIndex.end(); ++it) {
-        CBlockIndex* pindex = it->second;
+    for (auto& entry : mapBlockIndex) {
```

Comment:
Can be `const` (the loop modifies the entries pointed to by `mapBlockIndex`, not the entries directly.